node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -205,6 +212,52 @@ case class DataMessageHandler(
           s"Received headers=${headers.map(_.hashBE.hex).mkString("[", ",", "]")}")
         val chainApiF = chainApi.processHeaders(headers)
 
+        chainApiF.failed.map {
+          case r: RuntimeException
+              if r.getMessage.startsWith(
+                "Failed to connect any headers to our internal chain state") =>
```

Comment:
Do we want to do something more intelligent from the chain project on this? Perhaps rather than returning an exception in this case, return some type that allows you to pattern match on and handle the failure gracefully?

This also seems like it would be useful in the context of multiple peers if we have _seen_ a header before. So perhaps we would have an ADT

```scala

sealed trait ChainHeaderResult
object ChainHeaderResult {
  //we've successfully connect to header to our blockchain
  case object Success extends ChainHeaderResult
  //we've seen this header before
  case object Duplicate extends ChainHeaderResult
  //catch all for invalid headers, however we could break this down further to
  //bad POW, bad previous block hash, etc 
  case object Invalid extends ChainHeaderResult 
}
```